Dockerfile

```
@@ -1,8 +1,16 @@
+# First stage: Build the Go binaries
+FROM registry.redhat.io/rhel8/go-toolset:1.19.10-3 AS builder
+WORKDIR /app
+COPY . .
+RUN cd cmd/querygen && go build -o /app/build/querygen
```

Building this way would require changing the dockerfile each time we add a new binary, can we find a way to build the whole package with a  single command and have all the binaries be places in a single directory?

---

This should work (untested):
```suggestion
RUN cd cmd && go build -o /app/build ./...
```

---

yep this works (tested)

---

> Building this way would require changing the dockerfile each time we add a new binary, can we find a way to build the whole package with a single command and have all the binaries placed in a single directory?

@ifireball 
When you say add a new binary, do you mean the folder structure is expected to grow like below?

```
cmd/
  cmd1/
    main.go
  cmd2/
    main.go
  cmd3/
    main.go
  ...
```
In that case do we need to build binary with the name of folder name (Eg. cmd1, cmd2) and copy to the /usr/local/bin/?

---

yes, that kind of directory structure seems to be a convention for Go projects with multiple binaries. Hopefully it would simplify the task here

---

okay makes sense

---

> This should work (untested):

@amisstea @srivickynesh Ty.